pkg/os/windows/secpol/secpol_windows.go

```diff
@@ -0,0 +1,238 @@
+	 * see https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/security-policy-settings
+	 * and https://winaero.com/blog/reset-local-security-policy-settings-all-at-once-in-windows-10/
+	 */
+	sid, err := getSidOfUser(username)
+	if err != nil || sid == "" {
+		return false, fmt.Errorf("Unable to get user id: %v", err)
+	}
+
+	logonUserRight := hasLogonUserRight(sid)
+	if logonUserRight {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func hasLogonUserRight(userSid string) bool {
+	for _, sid := range getSidsWithLogonUserRights() {
+		if sid == userSid {
+			logging.Debug("Found sid in servicelogonright: ", sid)
+			return true
+		}
+	}
+	return false
+}
+
+// AllowUserToLogonAsService adds user's SID to 'seservicelogonright' config key in the security database
+func AllowUserToLogonAsService(username string) error {
+	userSid, err := getSidOfUser(username)
+	if err != nil {
+		return fmt.Errorf("Unable to get sid for username: %s: %v", username, err)
+	}
+
+	userLogonRightSids := getSidsWithLogonUserRights()
+	if len(userLogonRightSids) == 0 {
+		logging.Debugf("'seservicelogonright' config is empty or not present")
+	}
+
+	userSid = `*` + userSid
+
+	userLogonRightSids = append(userLogonRightSids, userSid)
+
+	seservicelogonrightConfig := fmt.Sprintf(config, strings.Join(userLogonRightSids, ","))
+
+	return configurePolicy(seservicelogonrightConfig)
+
+}
+
+func getSidsWithLogonUserRights() []string {
+	/* 'seservicelogon' key is a comma separated list of SIDs
+	 * the SIDs present are allowed to log on as a  service
+	 * seservicelogonright = *S-1-5-21-3938078522-2608308634-3841203849-1001,*S-1-5-80-0
+	 */
+	seservicelogonright := getRawConfig("seservicelogonright")
+	logging.Debug("Got raw config from file: ", seservicelogonright)
+	if len(seservicelogonright) == 0 {
+		return []string{}
+	}
+	var sids []string
+	s := strings.Split(seservicelogonright, "=")
+	if len(s) > 1 {
+		value := s[1]
+		value = strings.TrimSpace(value)
+
+		/* the sids in 'seservicelogonright' config have a * at the beginning
+		 * which needs to be removed to match with user sid
+		 */
+		value = strings.ReplaceAll(value, "*", "")
+		for _, s := range strings.Split(value, ",") {
+			if len(s) != 0 {
+				sids = append(sids, s)
+			}
+		}
+	}
+	return sids
+}
+
+func getRawConfig(configName string) string {
+	/* security configurations are stored in a inf file
+	 * each line contains a config key and a value delimited by an '=' sign
+	 * the value is a list of comma separated strings
+	 */
+
+	securityInfoFilePath, err := exportConfigFromSecurityDbToFile()
+	if err != nil {
+		logging.Debugf("Unable to get export config: %v", err)
+		return ""
+	}
+
+	defer func() {
+		_ = goos.RemoveAll(filepath.Dir(securityInfoFilePath))
+	}()
+
+	// open the security config file and get the line containing the config for configName
+	content, err := os.ReadFileUTF16(securityInfoFilePath)
```

Comment:
I am not sure why we need to handle the encoding here...

Reply:
We need to, or else the text is malformed when read directly, https://github.com/code-ready/crc/pull/1297#discussion_r439995256

Reply:
Can you be more precies in the naming... as it is unclear now if this is `ReadFileAsUTF16LE` or `ReadFileUTF16BE`

Reply:
Yes, this should be `ReadFileUTF16LE`

Reply:
Done!